db/crud.go

```diff
@@ -695,10 +706,12 @@ func (db *Database) ImportDoc(docid string, body Body, isDelete bool, importCas
 	case base.ErrImportCasFailure:
 		// Import was cancelled due to CAS failure.
 		return nil, err
+	case base.ErrImportRevFailure:
+		// Import was cancelled due to Rev import failure.
+		return nil, err
```

Comment:
I think it might be sufficient to fall through to 'default' here.  If we do that, I don't think we need the additional 'Import+' logging above.